### PR TITLE
Upgrade ktlint 0.50.0 to 1.2.1

### PR DESCRIPTION
**Pull request by TWiStErRob**
* Closes https://github.com/cashapp/paparazzi/pull/1076

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -15,5 +15,4 @@ ij_kotlin_allow_trailing_comma = false
 ij_kotlin_allow_trailing_comma_on_call_site = false
 
 ktlint_standard_argument-list-wrapping = disabled
-ktlint_standard_value-argument-comment = disabled
```
Comment (jrodbx) [inline .editorconfig:17]: unnecessary because now default?  or something else?

File: sample/build.gradle [hunk 2]
```diff
@@ -33,7 +33,7 @@ dependencies {
 
 // https://github.com/diffplug/spotless/issues/1572
 tasks.withType(com.diffplug.gradle.spotless.SpotlessTask).configureEach {
-  dependsOn(tasks.withType(Test))
+  mustRunAfter(tasks.withType(Test))
```
Comment (jrodbx) [inline sample/build.gradle:36]: good catch

File: .editorconfig [hunk 3]
```diff
@@ -9,10 +9,11 @@ insert_final_newline = true
...
+ktlint_code_style = intellij_idea
 ij_kotlin_imports_layout = *,java.**,javax.**,kotlin.**,kotlinx.**,^
 ij_kotlin_allow_trailing_comma = false
 ij_kotlin_allow_trailing_comma_on_call_site = false
 
 ktlint_standard_argument-list-wrapping = disabled
 ktlint_standard_value-argument-comment = disabled
-ktlint_function_naming_ignore_when_annotated_with = Composable
\ No newline at end of file
+ktlint_function_naming_ignore_when_annotated_with = Composable
```
Comment (jrodbx) [inline .editorconfig:18]: windows editing?

File: .editorconfig [hunk 4]
```diff
@@ -9,10 +9,11 @@ insert_final_newline = true
 end_of_line = lf
 
 [*.{kt, kts}]
+ktlint_code_style = intellij_idea
```
Comment (jrodbx) [inline .editorconfig:12]: same behavior?

File: paparazzi-gradle-plugin/src/test/projects/compose/src/test/java/app/cash/paparazzi/plugin/test/LoggerLeakTest.kt [hunk 5]
```diff
@@ -37,5 +37,7 @@ class LoggerLeakTest {
   }
 
   @Composable
-  private fun ComposeContent() { throw Exception() }
+  private fun ComposeContent() {
+    throw Exception()
+  }
```
Comment (jrodbx) [inline paparazzi-gradle-plugin/src/test/projects/compose/src/test/java/app/cash/paparazzi/plugin/test/LoggerLeakTest.kt:42]: nit: is this ktlint suggestion or preference?

File: .editorconfig (hunk 3 above)
Comment (TWiStErRob) [reply .editorconfig:18]: No, just editing, IDE adds it automatically.

Note: all files (even .java-version) in the project have new line at end of file, why not .editorconfig?

File: .editorconfig (hunk 4 above)
Comment (TWiStErRob) [reply .editorconfig:12]: [See](https://pinterest.github.io/ktlint/latest/rules/code-styles/#:~:text=Starting%20from%20version%201.0%2C%20ktlint_official%20is%20the%20default%20code%20style) and [this](https://github.com/pinterest/ktlint/pull/2144/files#diff-4e705beb17bb15a0aafaa58ead16c9ec09bd02cf2c8a2fc2c14fb375a87ccab6L54) which landed in 1.0. This was causing all the problems in https://github.com/cashapp/paparazzi/pull/1183

File: paparazzi-gradle-plugin/src/test/projects/compose/src/test/java/app/cash/paparazzi/plugin/test/LoggerLeakTest.kt (hunk 5 above)
Comment (TWiStErRob) [reply paparazzi-gradle-plugin/src/test/projects/compose/src/test/java/app/cash/paparazzi/plugin/test/LoggerLeakTest.kt:42]: https://github.com/cashapp/paparazzi/pull/1409/commits/897198a85c16b702c13ee1606ddaa9a665db24cc

**Review by jrodbx — APPROVED** [review]

**Pull request merged**